Makefile

```diff
@@ -1,13 +1,39 @@
+VERSION         := 0.08
+TARGET          := $(shell uname -r)
+DKMS_ROOT_PATH  := /usr/src/msi_ec-$(VERSION)
+
 obj-m += msi-ec.o
 
+
+KERNEL_MODULES	:= /lib/modules/$(TARGET)
+
+ifneq ("","$(wildcard /usr/src/linux-headers-$(TARGET)/*)")
+# Ubuntu
+KERNEL_BUILD	:= /usr/src/linux-headers-$(TARGET)
+else
+ifneq ("","$(wildcard /usr/src/kernels/$(TARGET)/*)")
+# Fedora
+KERNEL_BUILD	:= /usr/src/kernels/$(TARGET)
+else
+KERNEL_BUILD	:= $(KERNEL_MODULES)/build
+endif
+endif
+
+
+all: modules
+
 modules:
-	make -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) modules
```

Comment:
Is this compliant?
https://docs.kernel.org/kbuild/modules.html#command-syntax

`/lib/modules/$(shell uname -r)/build` should point to your kernel's header, no matter what is your operating system.  E.g, for Debian:
```bash
> readlink -f /lib/modules/5.10.0-16-amd64/build
/usr/src/linux-headers-5.10.0-16-amd64
```

Reply:
yes it picks the headers path based on the distro

Reply:
My mistake. I did a bad quoted. Please see: https://github.com/sainAk/msi-ec/blob/056dff5a8e1cf66efdba74e6a3522ae8849435ac/Makefile#L10-L20

What is the benefit of doing it this way?

```bash
# Ubuntu 22.04
> readlink -f /lib/modules/5.15.0-46-generic/build 
/usr/src/linux-headers-5.15.0-46-generic
```
```
# Fedora 36
> readlink -f /lib/modules/5.18.18-200.fc36.x86_64/build
/usr/src/kernels/5.18.18-200.fc36.x86_64
```